### PR TITLE
Fix empty file attributes leading to high CPU usage

### PR DESCRIPTION
**Pull request by ffried**
This should fix the high CPU problem in version 3.5.0 (#990). The dictionary must not set to an empty dictionary. Just returning an empty dictionary in case it's nil now, which lets it re-initialize once the filePath is set.

@bpoplauschi We should probably release a 3.5.1 with this fix. As described in #990 it makes 3.5.0 useless for some people.

### CODE REVIEW

**Comment by OSSPolice** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any changes to library code should be reflected in the CHANGELOG. Please consider adding a note there about your change.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The library files were changed, but the tests remained unmodified. Consider updating or adding to the tests to match the library changes.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Classes/DDFileLogger.m [hunk 1]
```diff
@@ -762,7 +764,7 @@ - (NSTimeInterval)rollingFrequency {
 - (void)setRollingFrequency:(NSTimeInterval)newRollingFrequency {
     dispatch_block_t block = ^{
         @autoreleasepool {
-            self->_rollingFrequency = newRollingFrequency;
+            self->_rollingFrequency = MIN(kDDMaxRollingFrequencey, newRollingFrequency);
```
Comment (lolgear) [inline Classes/DDFileLogger.m]:
```
    NSTimeInterval interval  = LONG_LONG_MAX / NSEC_PER_SEC;
    NSTimeInterval intervalee  = LONG_LONG_MAX + 1; // xcode says about overflow
    NSTimeInterval intervaler = MIN(interval, intervalee); // value is below zero
```

My suggestion:

```
        NSTimeInterval interval  = LONG_LONG_MAX / NSEC_PER_SEC;
        NSTimeInterval intervalee  = LONG_LONG_MAX + 1;
        NSTimeInterval intervaler = MIN(interval, intervalee);
        NSLog(@"hah? %@", @(intervaler));
        NSLog(@"hah abs %@", @(fabs(intervaler)));
    }
    
    {
        NSTimeInterval interval = LONG_LONG_MAX / NSEC_PER_SEC;
        NSTimeInterval intervalee = LONG_LONG_MAX + 1;
        NSTimeInterval intervaler = MIN(interval, fabs(intervalee));
        NSLog(@"nice max %@", @(intervaler));
        NSLog(@"nice max abs %@", @(fabs(intervaler)));
    }
    
    {
        NSTimeInterval interval = LONG_LONG_MAX / NSEC_PER_SEC;
        NSTimeInterval intervalee = LONG_LONG_MIN - 1;
        NSTimeInterval intervaler = MIN(interval, fabs(intervalee));
        NSLog(@"nice min %@", @(intervaler));
        NSLog(@"nice min abs %@", @(fabs(intervaler)));
    }
```

Comment (lolgear) [reply Classes/DDFileLogger.m]: But... Could we just use dates instead? Or _IF_ we can dare, we can add optional linking to CoreMedia with CMTime structure inside.

**Comment by hhanesand** [comment]
Isn't there still a danger of this addition overflowing?

```objc-c
NSTimeInterval ti = [logFileCreationDate timeIntervalSinceReferenceDate];
ti += _rollingFrequency;
```

**Comment by ffried** [comment]
@hhanesand I'll look into it.

**Pull request merged**